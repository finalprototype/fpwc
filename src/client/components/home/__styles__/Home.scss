.smb2badge {
  display: block;
  margin-bottom: 12px;
}
