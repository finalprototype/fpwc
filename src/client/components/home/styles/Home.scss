.smb2badge {
  padding: 10px 20px;
  margin-bottom: 12px;
  border-radius: 20px;
  border: 10px outset #b72504;
  background-color: #6185f8;
  font-family: 'SuperPlumberBrothers';
  line-height: 1.2;
}

.smb2badgetitle {
  font-size: 32px;
}

.smb2badgesubtitle {
  font-size: 24px;
}
