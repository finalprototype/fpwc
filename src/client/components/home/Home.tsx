import React from "react";

import styles from "./styles/Home.scss";

const Home: React.FunctionComponent = () => {
  return (
    <div className={styles.container}>
      Hello Lucy
      <br/>
      ❤️
    </div>
  );
};

export default Home;
