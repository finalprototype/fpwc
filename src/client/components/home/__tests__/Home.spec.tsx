import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import Home from '../Home';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('Home', () => {
  it('renders', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches background update', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0);
    mount(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: 'BACKGROUND_CHANGE',
      payload: {
        imageSource: 'mock-file',
        videoSource: 'mock-file',
      },
    });
  });
});
