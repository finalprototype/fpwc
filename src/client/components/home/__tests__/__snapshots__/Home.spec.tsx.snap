// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<Fragment>
  <PageContent
    centered={true}
    flex={true}
    full={false}
  >
    <img
      alt="Super Mario Bros. 2 - Javascript Edition"
      className="smb2badge"
      src="mock-file"
    />
    Currently in development.
    <br />
    In the meantime, checkout the 
    <Link
      to="/smb"
    >
      SMB1 prototype
    </Link>
  </PageContent>
  <Footer
    fixed={true}
  />
</Fragment>
`;
